pr_parser.py

```diff
@@ -217,100 +158,119 @@ def get_all_related_prs(self, repo, sha, pull_id, actual_commit):
+        gh = self.__gh
+        branch = gh.get_repo(repo).get_branch(branch)
+        latest_commit = branch.commit.sha
+        return latest_commit
 
-                    #if on-core and on-tasks coexist, because the on-http and on-taskgraph depend both the two repos
-                    #so in their unit-test both the two commits should be provided, so extra information of on-tasks 
-                    #will be passed to on-core 
-                    if properties_dic.has_key('ON_TASKS'):
-                        properties_dic['ADD_FORKURL'] = 'https://github.com/changev/on-tasks.git'
-                        properties_dic['ADD_COMMIT'] = '{0}'.format(properties_dic["ON_TASKS_ghprbActualCommit"])
-
-        for k,v in properties_dic.iteritems():
-            properties += "{0}={1}\n".format(k, v)
-        
-        print "INFO: properties generation done"
-        return properties
-
-
-    def write_build_properties_file(self):
+    def wrap_manifest_file(self, file_path):
         """
-        write build properties file, this likes a main function
+        Generated manifest file
         """
-        all_prs = self.get_all_related_prs(self.__repo, self.__merge_commit_sha, self.__pull_id, self.__actual_commit)
-        properties = self.generate_build_properties(all_prs, self.__target_branch)
-
-        #Jenkins:ingore in root pr -> get_all_related_prs return [] -> no trigger downstream and set commit status "pending"
-        if not all_prs:
-            file_name = "ignore"
-        else:
-            repo_name = self.__repo.split('/')[1]
-            is_multi = bool(len(all_prs)>1)
-            #IMPORTANT! In Jenkins the existence of on-xxx file is the trigger of downstream jobs
-            file_name = "on-multi" if is_multi else repo_name
-
-        try:
-            with open(file_name, 'w') as target_file:
-                target_file.write(properties)
-        except IOError as error:
-            print "ERROR: file {0} does not exist.\n{1}".format(file_name, error)
-            sys.exit(1)
-        print "INFO: build properties file **{0}** has been created".format(file_name)
-        return
-        
+        all_prs = self.get_all_related_prs(self.__repo, self.__merge_commit_sha, self.__pull_id)
+        under_test_prs = self.get_under_test_prs()
+        # instance of manifest template
+        manifest = Manifest.instance_of_sample("manifest-pr-gate.json")
+
+        # wrap with pr
+        repo_url_list = [repo["repository"] for repo in manifest.repositories]
+        for pr in all_prs:
+            repo, sha1, _ = pr
+            repo_url = "https://github.com/{0}.git".format(repo)
+            # uniform the repo_url case, make sure the url is completely consistent with repo in the manifest
+            repo_url = [url for url in repo_url_list if url.lower()== repo_url][0]
+            if repo in under_test_prs:
+                manifest.update_manifest(repo_url, "", sha1, True)
```

Comment:
could it be  manifest.update_manifest(repo_url, "", sha1,  repo in under_test_prs) ?